### ISSUE TITLE
google_project_services Not Recognizing Dependent Services

### ISSUE THREAD

**Issue by callensm** [terraform-providers/terraform-provider-google#1658]
Hi there,

I've run into an issue using the `google_project_services` resource where Terraform doesn't recognize services that are enabled as a bi-product of enabling others (i.e. having service dependencies).

In my particular case, as a result of enbabling the `cloudapis.googleapis.com` service, the project also enables all other services that come along or are associated with this. This is fine, however, when I run another `terraform apply`, the resource doesn't seem to understand that these were enabled as dependent services and attempts to shut them all down, resulting in a circular chain of shutting down and booting up services that it doesn't recognize.

### Terraform Version
```sh
~ terraform -v
Terraform v0.11.7
```

### Affected Resource(s)
- google_project_services

### Terraform Configuration Files

_Arbitrary values given for project and folder IDs as they aren't relevant_
```hcl
provider "google" {
  region = "us-east1"
}

resource "google_project" "test" {
  name        = "Test Project"
  project_id = "test-project-12345"
  folder_id   = "1234567890"

  labels {
    terraform = "true"
  }
}

resource "google_project_services" "enabled_services" {
  project   = "${google_project.test.project_id}"
  services = ["cloudapis.googleapis.com", "dialogflow.googleapis.com"]
}
```

### Debug Output
![e8c8fdc24cf2c441acab4439ed58186e namprd06 prod outlook com](https://user-images.githubusercontent.com/13121516/41427039-a4f992ec-6fd3-11e8-88ed-53b841b929a4.png)
![fa83e4fb0c037b45832e3dd379deee55 namprd06 prod outlook com](https://user-images.githubusercontent.com/13121516/41427040-a504015a-6fd3-11e8-8db8-f8372c026298.png)

### Expected Behavior
Terraform enables all services that are associated with those specified in the `google_project_services` resource declaration, and doesn't shut down additional service dependencies on the next `apply`

### Actual Behavior
Shuts down additionally enabled services on the second `apply` that were enabled as a product of those specified in the Terraform script.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Notice that 14 services are now enabled instead of the 2 specified because they are dependents
3. `terraform apply`
4. Notice that Terraform attempts to shut down nearly all services that were enabled

**Comment by danawillow** [comment]
Hey @callensm, this is the exact use case for the `google_project_service` resource. Have you looked into that yet? https://www.terraform.io/docs/providers/google/r/google_project_service.html

**Comment by callensm** [comment]
@danawillow Yes I have. My understanding is that `google_project_service` is used for enabling a _single_ service on a specific project, while `google_project_services` is for enabling _multiple_ services for a project.

My intent _is_ to enable multiple services as shown in the sample Terraform script, however, the result of enabling the `cloudapis.googleapis.com` service is an additional 12-13 services that are contained within that definition, yet Terraform doesn't recognize that and tears them down on the next `apply`.

Unless I'm completely missing or misunderstanding something, I hope that clarifies the bug.

**Comment by danawillow** [comment]
You're mostly right. `google_project_services` is for enabling an **authoritative list** of services for a project. So, if a service enables another as a side effect and that one doesn't appear in the list, Terraform will try to destroy it because you didn't say you wanted the other service. `google_project_service` is for enabling a single service, without any regard to what other services you have enabled. So if your goal is to enable cloudapis and dialogflow, you can create two `google_project_service` resources (one for each), and then Terraform won't worry about the side effects- it'll do your intent, which was to to ensure that those two services were enabled.
